drivers/platform/maxim/maxim_rtc.c

```diff
@@ -0,0 +1,315 @@
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/************************* Include Files **************************************/
+/******************************************************************************/
+
+#include <stdlib.h>
+#include <errno.h>
+#include "no-os/irq.h"
+#include "no-os/rtc.h"
+#include "no-os/util.h"
+#include "rtc.h"
+#include "rtc_extra.h"
+#include "rtc_regs.h"
+
+#define MS_TO_RSSA(x) (0 - ((x * 256) / 1000))
+
+/**
+* @brief Callback descriptor that contains a function to be called when an
+* interrupt occurs.
+*/
+static struct callback_desc *cb;
+
+/******************************************************************************/
+/************************ Functions Definitions *******************************/
+/******************************************************************************/
+
+void RTC_IRQHandler()
+{
+
+	mxc_rtc_regs_t *rtc_regs = MXC_RTC;
+	volatile uint32_t rtc_ctrl = rtc_regs->ctrl;
+	uint8_t n_int = 0;
+	/** Sub-second alarm flag clear */
+	rtc_regs->ctrl &= ~BIT(7);
+	/** Time-of-day alarm flag clear */
+	rtc_regs->ctrl &= ~BIT(6);
+	/** RTC (read) ready flag */
+	rtc_regs->ctrl &= ~BIT(5);
+
+	if(!cb)
+		return;
+
+	/** Shift right so the interrupt flags will be the first 3 bits */
+	rtc_ctrl >>= 5UL;
+	/** Clear the remaining bits */
+	rtc_ctrl &= 0x7UL;
+	while(rtc_ctrl) {
+		n_int = find_first_set_bit(rtc_ctrl);
+		if (rtc_ctrl & (rtc_regs->ctrl & BIT(n_int))) {
+			cb->callback(cb->ctx, n_int, cb->config);
+		}
+		rtc_ctrl >>= n_int + 1;
+	}
+}
+
+/**
+ * @brief Initialize the RTC peripheral.
+ * @param device - The RTC descriptor.
+ * @param init_param - The structure that contains the RTC initialization.
+ * @return 0 in case of success, errno codes otherwise.
+ */
+int32_t rtc_init(struct rtc_desc **device, struct rtc_init_param *init_param)
+{
+	int32_t ret = 0;
+	struct rtc_desc *dev;
+	sys_cfg_rtc_t sys_cfg;
+
+	if(!init_param)
+		return -EINVAL;
+
+	struct rtc_init_maxim *maxim_init_param = init_param->extra;
+	dev = calloc(1, sizeof(*dev));
+	if (!dev)
+		return -ENOMEM;
+
+	dev->id = init_param->id;
+	dev->freq = init_param->freq;
+	dev->load = init_param->load;
+
+	TMR_Enable(MXC_TMR0);
+
+	if (RTC_Init(MXC_RTC, dev->load, maxim_init_param->ms_load,
```

Comment:
We could use an extra rtc descriptor for maxim to store the `MXC_RTC` element. This way the driver is scalable to maxim devices with more than one RTC.